### DISCUSSION TITLE
feat: Secret Manager Support

### DISCUSSION THREAD

**Discussion by kaddybrar** [awslabs/aws-mysql-jdbc#183] [Ideas]
Does aws-mysql-jdbc provide similar support for retrieving Database UserName and Password from AWS secret-manager? We would need this feature to enable the successful jdbc-credentials refresh in case of password rotations.
This is something com.amazonaws.secretsmanager.sql.AWSSecretsManagerMariaDBDriver provides.

**Comment by hsuamz** [comment]
Hello @kaddybrar,

Thanks for reaching out. To clarify, are you looking for official support through this [project](https://github.com/aws/aws-secretsmanager-jdbc) with the driver?

Thanks,

Matthew

**Reply by kaddybrar** [reply]
@hsuamz yes this the feature we need in this driver as well.

**Reply by kaddybrar** [reply]
following up on this. Since, the latest version of the MariaDB JDBC Driver (Connector/J) no longer supports Amazon Aurora, this feature request is critical to us for onboarding our services to use this driver

**Reply by hsuamz** [reply]
Hello @kaddybrar,

We're currently looking into Secret Manager Support for the driver. As we've implemented a plugin manager for the driver, we're looking into the possibility of adding a Secret Manager plugin to perform authentication.

We're also reaching out to the team in the above response to determine how we can support the driver in their project.

We cannot confirm any dates at this time, but can confirm that we're looking into this. Since we work out in the open, we'll update this discussion with implementation visibility when we can.

Thanks,

Matthew

**Comment by hsuamz** [comment]
Following up regarding this question, a new version of the driver has been released that has support for Secrets Manager. You can find the release here: https://github.com/awslabs/aws-mysql-jdbc/releases/tag/1.1.0.

**Comment by hsuamz** [comment]
Secrets Manager support has been officially released.
